profiling/src/internal/profile/fuzz_tests.rs

```diff
@@ -0,0 +1,631 @@
+            );
+            let owned_function = Function::new(
+                profile
+                    .string_table_fetch(function.name)
+                    .clone()
+                    .into_boxed_str(),
+                profile
+                    .string_table_fetch(function.system_name)
+                    .clone()
+                    .into_boxed_str(),
+                profile
+                    .string_table_fetch(function.filename)
+                    .clone()
+                    .into_boxed_str(),
+                function.start_line,
+            );
+            let owned_location =
+                Location::new(owned_mapping, owned_function, location.address, line.line);
+
+            owned_locations.push(owned_location);
+        }
+
+        // Recreate owned_labels from vector of pprof::Label
+        let mut owned_labels = Vec::new();
+        for label in sample.labels.iter() {
+            let key = profile
+                .string_table_fetch(label.key)
+                .clone()
+                .into_boxed_str();
+
+            if *key == *"end_timestamp_ns" {
+                // TODO: Check end timestamp label
+                continue;
+            } else if *key == *"trace endpoint" {
+                let actual_str = profile.string_table_fetch(label.str);
+
+                let prev_label: &Label = owned_labels
+                    .last()
+                    .expect("Previous label to exist for endpoint label");
+                let num = prev_label.num as u64;
+                let expected_str = endpoint_mappings
+                    .get(&num)
+                    .expect("Endpoint mapping to exist");
+
+                assert_eq!(actual_str, *expected_str);
+                continue;
+            }
+
+            if label.str != 0 {
+                let str = Box::from(profile.string_table_fetch(label.str).as_str());
+                owned_labels.push(Label {
+                    key,
+                    str: Some(str),
+                    num: 0,
+                    num_unit: None,
+                });
+            } else {
+                let num = label.num;
+                let num_unit = if label.num_unit != 0 {
+                    Some(
+                        profile
+                            .string_table_fetch(label.num_unit)
+                            .clone()
+                            .into_boxed_str(),
+                    )
+                } else {
+                    None
+                };
+                owned_labels.push(Label {
+                    key,
+                    str: None,
+                    num,
+                    num_unit,
+                });
+            }
+        }
+
+        if let Some(expected_sample) = expected_timestamped_samples.next() {
+            assert_eq!(owned_locations, expected_sample.locations);
+            assert_eq!(sample.values, expected_sample.values);
+            assert_eq!(owned_labels, expected_sample.labels);
```

Comment:
These should potentially be sorted: there is no requirement that labels remain in the same order

Reply:
Locally I'm sorting the labels and got a new failure:

```
======================== Test Failure ========================

Input: 
(
    [],
    [
        (
            None,
            Sample {
                locations: [],
                values: [],
                labels: [
                    Label {
                        key: "local root span id",
                        str: None,
                        num: 281474976710656,
                        num_unit: None,
                    },
                    Label {
                        key: "",
                        str: Some(
                            "",
                        ),
                        num: 0,
                        num_unit: Some(
                            "",
                        ),
                    },
                ],
            },
        ),
    ],
)

Error: 
panicked at profiling/src/internal/profile/fuzz_tests.rs:402:18:
Value not found for an aggregated sample
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

==============================================================
```

Line 402 is this expect:

```rs
            let key = (&owned_locations, &owned_labels);
            let expected_values = samples_without_timestamps
                .get(&key)
                .expect("Value not found for an aggregated sample");
            assert_eq!(&sample.values, expected_values);
```

I think this happens because `Vec<Label>` and such are used as keys in the hash, and then they don't work out because of order?

Reply:
I pushed this up as `fuzz_failure_001`.